frontend/src/app/containers/ExperimentPage.js

```diff
@@ -649,19 +650,34 @@ export class ExperimentDetail extends React.Component {
     // Ignore subsequent clicks if already in progress
     if (isEnabling) { return; }
 
+    let installAddonPromise = Promise.resolve();
+    if (! this.props.hasAddon) {
+      let {installAddon, requireRestart, sendToGA, eventCategory} = this.props;
+
+      installAddonPromise = installAddon(
+        requireRestart,
+        sendToGA,
+        eventCategory);
+    }
+
     this.setState({
       isEnabling: true,
       isDisabling: false,
-      shouldShowTourDialog: true,
-      progressButtonWidth: evt.target.offsetWidth
+      shouldShowTourDialog: true
     });
 
-    enableExperiment(experiment);
-
-    sendToGA('event', {
-      eventCategory: 'ExperimentDetailsPage Interactions',
-      eventAction: 'Enable Experiment',
-      eventLabel: experiment.title
+    installAddonPromise.then(() => {
+      // XXX installAddonPromise shouldn't resolve until the add-on is ready
+      // to accept messages, instead of using a racy setTimeout here.
+      setTimeout(() => {
```

Comment:
The right way to do this would be to poll for the existence of the [`navigator.testpilotAddon` flag](https://github.com/mozilla/testpilot/blob/e209f400492e6ddc87aae15814d36447ba50569c/addon/data/set-installed-flag.js#L11), which the add-on adds on init [in a PageMod](https://github.com/mozilla/testpilot/blob/276d2021bbd474309e37971d62a7d4b07d6f2a5b/addon/lib/app.js#L43-L50).

The promise might look like this (rejecting after 10 seconds under the assumption that something went wrong; see #660,  #1227):

```js
return new Promise((resolve, reject) => {
  let i = 0;
  const interval = setInterval(() => {
    i++;
    if (window.testpilotAddon) {
      clearInterval(interval);
      resolve();
    } else if (i > 100) {
      clearInterval(interval);
      reject();
    };
  }, 100);
});
```